main.js

```diff
@@ -1,123 +1,169 @@
+
+    anchorageID = channelDetails.nextAnchorageID;
+  } else {
+    channelDetails = {};
+
+    // A new channel is bound and created. Node not specified -> Chrysalis mainnet
+    channel = await IotaAnchoringChannel.bindNew(options);
+
+    channelDetails.channelID = channel.channelID;
+    // This seed will also be used later to anchor more messages to the channel
+    channelDetails.seed = channel.seed;
+    // In our case Author Pub Key === Subscriber Pub Key
+    channelDetails.publicKey = channel.subscriberPubKey;
+
+    anchorageID = channel.firstAnchorageID;
   }
-};
+
+  const anchoringResult = await channel.anchor(
+    Buffer.from(sha256),
+    anchorageID
+  );
+
+  // The anchorage that will be used for the next one
+  const nextAnchorageID = anchoringResult.msgID;
+  channelDetails.nextAnchorageID = nextAnchorageID;
+
+  return channelDetails;
+}
 
 /**
- * Update the target with the IOTA MAM root, if it does not exist, and always update the MAM state.
+ * Update the target with the channel details
  *
- * @param {object} data - root and mamState from sendToIOTA().
- * @returns {Promise} Promise that resolves once the target is updated.
+ * @param {object} target the item
+ * @param {string} targetType the type of item (thng, product, etc.)
+ * @param {object} channelDetails - anchoring channel details from sendToIOTA().
+ * 
+ * @returns {object} the updated target
  */
-const updateTarget = (data) => {
-  if (!target.customFields) {
-    target.customFields = {};
-  }
+async function updateTarget (target, targetType, channelDetails) {
+  const customFields = target.customFields || {};
 
-  const customFields = Object.assign(target.customFields, { iotaMamState: data.mamState });
-  if (!target.customFields.iotaRoot) {
-    customFields.iotaRoot = data.root;
-  }
+  // We overwrite or assign the channel details
+  customFields.iotaAnchoringChannel = channelDetails;
 
-  return app[targetType](target.id).update({ customFields });
-};
+  const result = await app[targetType](target.id).update({ customFields });
+
+  return result;
+}
 
 /**
- * Create a confirmation action containing the original action and the IOTA root
+ * Create a confirmation action containing the original action
+ *  and the IOTA anchoring channel ID
  *
- * @returns {Promise} Promise that resolves to the confirmation action.
+ * @param {object} action The concerned action
+ * @param {object} target The target on which the action was executed
+ * @param {string} targetType the type of item (thng, product, etc.)
+ * 
+ * @returns {object} the confirmation action
+ * 
  */
-const createConfirmation = () => {
+async function createConfirmation (action, target, targetType) {
   const payload = {
     type: CONFIRMATION_ACTION_TYPE,
     [targetType]: action[targetType],
     customFields: {
-      originalAction: action,
-      iotaRoot: target.customFields.iotaRoot,
-    },
+      originalAction: {
```

Comment:
Previous projects relied heavily upon the _entire_ original action being available here, not just `type` and `id`. 

Specifically because they were created by different users and could not all be read by the consuming app - so all details must be available here without a requirement to do a lookup on action `id`.

Reply:
Unless of course those projects are dead and new ones won't have that requirement for App Users to lookup actions.

Reply:
that can be restored but If an application by definition can not read the action why it could do it if embedded here? Why is it needed to duplicate the info? is there a better architectural solution? 

@domguinard